go/arrow/csv/common.go

```diff
@@ -223,6 +223,7 @@ func validate(schema *arrow.Schema) {
 		case *arrow.Date32Type, *arrow.Date64Type:
 		case *arrow.Decimal128Type, *arrow.Decimal256Type:
 		case *arrow.ListType:
+		case arrow.BinaryDataType:
```

Comment:
if you're not going to support the `LargeString` and `LargeBinary` types, then this should be just `*arrow.BinaryType` instead of `arrow.BinaryDataType` which is an interface that includes the Large variants